### PR TITLE
feat: Script for starting a local Regen test node, based on the tutorial

### PR DESCRIPTION
**Pull request by frumioj**
## Description

Added a script that runs most of the commands shown in the tutorial https://docs.regen.network/getting-started/ to make starting a node quicker than running the commands individually. If a command fails, the script errors and ends.

Two command-line options are allowed for specifying a keyring-backend (-k), and the name of the blockchain (-c). 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
I'm not sure if we need this since there's already some `testnet` command built-in cosmos-sdk

**Comment by frumioj** [comment]
> I'm not sure if we need this since there's already some `testnet` command built-in cosmos-sdk

I merely intended to duplicate what is done in the Regen "Getting Started" doc (https://docs.regen.network/getting-started/). I'd also note that I personally very often start over with a brand new node configuration, so I was often repeating the same steps while doing development.

If the `testnet` route is the way, perhaps this should be noted in the "Getting Started" doc?

**Comment by ryanchristo** [comment]
Thanks for the pull request @frumioj!

> If the testnet route is the way, perhaps this should be noted in the "Getting Started" doc?

Yes, this should be noted in "Getting Started" but the `testnet` command is not yet implemented. I opened #563 for tracking. We could use this script in the meantime.

File: scripts/start_testnode.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Get the options
+while getopts ":kc:" option; do
+   case $option in
+      k) # Keyring backend
+         KeyringBackend=$OPTARG;;
+      c) # Enter a name
+         Chain=$OPTARG;;
+     \?) # Invalid option
+         echo "Error: Invalid option"
+         exit;;
```
Comment (ryanchristo) [inline scripts/start_testnode.sh:21]:
```suggestion
         exit 1
```

File: scripts/start_testnode.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+regen keys add validator --keyring-backend ${KeyringBackend} || exit_with_error "Error: Validator add failed"
+regen keys add delegator --keyring-backend ${KeyringBackend} || exit_with_error "Error: Delegator add failed"
+regen init node --chain-id ${Chain} || exit_with_error "Error: Could not init node"
+
+# Change the staking token to uregen
+# Note: sed works differently on different platforms
+echo "Updating your staking token to uregen in the genesis file..."
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
```
Comment (ryanchristo) [inline scripts/start_testnode.sh:38]:
```suggestion
if [[ `uname` == "Linux"* ]]; then
```

File: scripts/start_testnode.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    echo "Your OS is a Linux variant..."
+    sed -i "s/stake/uregen/g" ~/.regen/config/genesis.json || exit_with_error "Error: Could not update staking token"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (ryanchristo) [inline scripts/start_testnode.sh:41]:
```suggestion
elif [[ `uname` == "Darwin"* ]]; then
```

File: scripts/start_testnode.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    echo "Your OS is Mac OS/darwin..."
+    sed -i "" "s/stake/uregen/g" ~/.regen/config/genesis.json || exit_with_error "Error: Could not update staking token"
+else
+    # Dunno
+    echo "Your OS is not supported"
+    exit ;;
```
Comment (ryanchristo) [inline scripts/start_testnode.sh:47]:
```suggestion
    exit 1
```

**Review by ryanchristo — COMMENTED** [review]
I'm receiving the following error when running the script:

```
./scripts/start_testnode.sh: 47: Syntax error: ";;" unexpected (expecting "fi")
```

It looks like two instances of `exit ;;` should be `exit 1`.

Also, `$OSTYPE` is not working for me on the latest version of Ubuntu (I receive `linux-gnu: command not found` when checking `$OSTYPE`). After doing a little research, it looks like and using `uname` instead of `$OSTYPE` might be the best solution.

Otherwise this looks good. I wouldn't be opposed to including it as a temporary solution until we have the `testnet` command up and running.

**Comment by frumioj** [comment]
Thanks @ryanchristo 👍

**Review by ryanchristo — APPROVED** [review]
tACK

```
bash ./scripts/start_testnode.sh
```

One more request to update `#!/bin/sh` to `#!/bin/bash`. We use `#!/bin/bash` in other scripts and it will eliminate the need to specify `bash` when running the command.

Pre-approving. Thanks again!

File: scripts/start_testnode.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (frumioj) [reply scripts/start_testnode.sh]: Done - thanks again @ryanchristo!

**Comment by blushi** [comment]
> Thanks for the pull request @frumioj!
> 
> > If the testnet route is the way, perhaps this should be noted in the "Getting Started" doc?
> 
> Yes, this should be noted in "Getting Started" but the `testnet` command is not yet implemented. I opened #563 for tracking. We could use this script in the meantime.

Right, I didn't realized this was only done on cosmos-sdk simapp!

**Review by blushi — APPROVED** [review]
Thanks @frumioj!

**Pull request merged**